core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -611,6 +611,35 @@ protected T loadValue() {
     }
   }
 
+  /**
+   * Created a gauge that aggregates the value of existing gauges.
+   *
+   * @param name the gauge name
+   * @param metrics the set of metric values to be aggregated
+   * @param timeout the cached gauge timeout
+   * @param timeUnit the unit of timeout
+   */
+  public static synchronized void registerAggregatedCachedGaugeIfAbsent(
+      String name, Set<MetricKey> metrics, long timeout, TimeUnit timeUnit) {
+    if (METRIC_REGISTRY.getMetrics().containsKey(name)) {
+      return;
+    }
+    METRIC_REGISTRY.register(name, new CachedGauge<Double>(timeout, timeUnit) {
+      @Override
+      protected Double loadValue() {
+        double total = 0.0;
+        for (MetricKey key : metrics) {
+          Metric m = getMetricValue(key.getName());
```

Comment:
Maybe there is a cheaper way to get the value, since this creates a new object, you can probably just get the value directly?

Reply:
The object is created to make sure the metrics being summed exist and are of the right type. Getting the value directly could cause `NullPointerException` or other incompatible value problems.

Reply:
I mean in ``getMetricValue`` calls ``getAlluxioMetricFromCodahaleMetric`` which calls ``Metric.from`` using a number value as input which allocates and calculates a lot of things. But then just the number value is used. I wonder why not just use this number directly without calling ``Metric.from``? This may require a small refactor of ``getAlluxioMetricFromCodahaleMetric`` though. It should be ok to sum the values of any type of metric that is a number also? eg. the user may want to sum a set of timers for example.